pkg/detectors/aws/aws.go

```diff
@@ -101,6 +103,34 @@ func GetHMAC(key []byte, data []byte) []byte {
 	return hasher.Sum(nil)
 }
 
+func GetAccountNumFromAWSID(AWSID string) (string, error) {
+	// Function to get the account number from an AWS ID (no verification required)
+	// Source: https://medium.com/@TalBeerySec/a-short-note-on-aws-key-id-f88cc4317489
+	if len(AWSID) < 4 {
+		return "", fmt.Errorf("AWSID is too short")
+	}
+	trimmed_AWSID := AWSID[4:]
+	x, err := base32.StdEncoding.WithPadding(base32.NoPadding).DecodeString(strings.ToUpper(trimmed_AWSID))
+	if err != nil {
+		return "", err
+	}
+
+	if len(x) < 6 {
+		return "", fmt.Errorf("Decoded AWSID is too short")
+	}
+	y := x[0:6]
+
+	var z uint64 = binary.BigEndian.Uint64(append(make([]byte, 8-len(y)), y...))
+	maskBytes, err := hex.DecodeString("7fffffffff80")
```

Comment:
suggestion:  Instead of storing the hexadecimal string, we can calculate the uint64 mask value directly and store that as a constant. This means we eliminate not only the hex decoding in each function call but also the byte slice to uint64 conversion.

Ex:
`const mask uint64 = 0x7fffffffff80`

Reply:
Forgot to drop the benchmark differences.
![Screenshot 2023-11-06 at 6 38 48 AM](https://github.com/trufflesecurity/trufflehog/assets/21311841/eb461e64-7dae-45c4-912b-7eefd8693b1c)